src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -152,6 +145,13 @@ func (s *peersSource) Read(
 	s.log.Info("bootstrapping time series data success",
 		zap.Duration("took", nowFn().Sub(start)))
 
+	alloc := s.opts.ResultOptions().IndexDocumentsBuilderAllocator()
+	segBuilder, err := alloc()
+	if err != nil {
+		return bootstrap.NamespaceResults{}, err
+	}
+	builder := result.NewIndexBuilder(segBuilder)
```

Comment:
Do we want to alloc a new builder every Read call? Should we pool these?

Reply:
That's actually what we're hoping to do, a single bootstrap for given timee window per only calls Read once, the bootstrapper was holding onto it indefinitely leaking some memory.